sleap/gui/web.py

```diff
@@ -146,6 +149,49 @@
         )
 
 
+@attr.s(auto_attribs=True)
+class AnnouncementChecker:
+    """Checker for new announcements on the bulletin page of sleap."""
+
+    bulletin_json_path: str = BULLETIN_JSON
+    previous_announcement_date: str = prefs.prefs["announcement last seen date"]
+
+    def check_for_announcement(self) -> bool:
+        """Returns if new announcements are available."""
+        try:
+            # Attempt to open the file in read mode
+            with open(self.bulletin_json_path, 'r', encoding='utf-8') as jsf:
+                # Load the JSON data into a Python data structure
+                data = json.load(jsf)
+                latest_data = data[0]
+                
+                if latest_data['date'] != self.previous_announcement_date:
+                    return True
+        except FileNotFoundError:
+            return False
+
+    def get_latest_announcement(self) -> Optional[Tuple[str, str]]:
+        """Return latest announcements on the releases page not seen by user."""
+        success = self.check_for_announcement()
+        if success:
+            # Attempt to open the file in read mode
+            with open(self.bulletin_json_path, 'r', encoding='utf-8') as jsf:
+                # Load the JSON data into a Python data structure
+                data = json.load(jsf)
+                latest_data = data[0]
+                
+                if latest_data['date'] != self.previous_announcement_date:
+                    return (latest_data['date'], latest_data['content'])
+        return None
```

Comment:
Not a bad idea... maybe we can create a helper function just for reading the data and only use this reading function once in `check_for_announcement` where we can store the `data` in an attribute for access between both functions?

Reply:
In this case where are we calling the `check_for_announcement' function? Since we don't call the function in the `get_latest_announcement`, `self._latest_data` would be `None` always. So, can we just have one function that checks and returns an announcement if there is one? Or do you suggest something else.

Reply:
@coderabbitai what do you think of my previous comment?

Reply:
@roomrys does this work, or do we need to store the bulletin JSON file as an attribute after reading it?

Reply:
I think this simplifies things, no need to have something (`check_for_announcement`) that we don't really need.